akka-actor-typed/src/main/scala/akka/actor/typed/Entity.scala

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2021 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.actor.typed
+
+import scala.reflect.ClassTag
+
+import akka.actor.InvalidMessageException
+import akka.actor.WrappedMessage
+import akka.actor.typed.Entity.DataCenter
+import akka.actor.typed.Entity.EntityAllocationStrategy
+import akka.actor.typed.Entity.EntityCommand
+import akka.actor.typed.Entity.EntityMessageExtractor
+import akka.actor.typed.Entity.EntitySettings
+import akka.annotation.DoNotInherit
+import akka.annotation.InternalApi
+
+object Entity {
+
+  @DoNotInherit trait EntityCommand
+
+  final case class Passivate[M](entity: ActorRef[M]) extends EntityCommand
+
+  type DataCenter = String
+  trait EntitySettings
+  trait EntityMessageExtractor[E, M]
+  trait EntityAllocationStrategy
+
+  def apply[M](typeKey: EntityTypeKey[M])(
+      createBehavior: EntityContext[M] => Behavior[M]): Entity[M, EntityEnvelope[M]] =
+    new Entity(createBehavior, typeKey, None, Props.empty, None, None, None, None, None)
+
+  final case class EntityEnvelope[M](entityId: String, message: M) extends WrappedMessage {
+    if (message == null) throw InvalidMessageException("[null] is not an allowed message")
+  }
+}
+
+final class Entity[M, E] private[akka] (
+    val createBehavior: EntityContext[M] => Behavior[M],
+    val typeKey: EntityTypeKey[M],
+    val stopMessage: Option[M],
+    val entityProps: Props,
+    val settings: Option[EntitySettings],
+    val messageExtractor: Option[EntityMessageExtractor[E, M]],
+    val allocationStrategy: Option[EntityAllocationStrategy],
+    val role: Option[String],
+    val dataCenter: Option[DataCenter])
```

Comment:
There are several things that are only relevant in a clustered setup. role and dataCenter obviously, but also the whole idea of shard and allocationStrategy are not relevant for the local case.

Reply:
One idea is to only support the very basic `init` in akka-actor-typed. No custom envelope, no custom message extractor, no allocationStrategy. For more advanced cases the `init` in akka-cluster-sharding-typed` can be used. 

That means that only `EntityTypeKey` and `EntityRef` have to be moved to `akka-actor-typed`.

Reply:
Yes, everything cluster/sharding related could be removed. But that also means that when moving to sharded entities, the user will need to adapt. 

Also, what would that mean to declare an entity using the local API, but on a clustered system? Will it be unique on that node only? Or will it be a sharded one? 

So maybe we should take this as a trade-off. We will have some settings that only make sense when clustered and when used in a non-clustered system they are simply ignored. 

If we chose to have new APIs and deprecate the exiting sharding entity, then we could make it more explicitly as part of a cluster API. Something like:

```scala
Entity(...).withShardingSettings(...)
``` 

When not specified, fallback to some defaults. Actually, pretty much like now with `Option` but behind an explicit sharding settings.

Reply:
> But that also means that when moving to sharded entities, the user will need to adapt.
>
> Also, what would that mean to declare an entity using the local API, but on a clustered system? Will it be unique on that node only? Or will it be a sharded one?

I was thinking that the basic `init` API would also work for a clustered system, and then it would `init` ClusterSharding under the hood. All these options are optional in ClusterSharding.

Meaning that the API would only change if user needs something more advanced, then they would have to use the ClusterSharding.init instead.